### ISSUE TITLE
Replace external actions in CI

### ISSUE THREAD

**Issue by simonsan** [rust-unofficial/patterns#244]
As preparation for a possible inclusion into `rust-lang` we should replace the external actions we use with e.g. scripts that do the same.

https://github.com/rust-lang/std-dev-guide/pull/7#issuecomment-770317395

That PR also shows how that could be done:
https://github.com/rust-lang/std-dev-guide/pull/7/files

**Comment by MarcoIeni** [comment]
I think I would do it only immediately before moving the book under rust-lang.
Implementing all the actions by ourselves means that we have to maintain them in the meantime. For now we can save that effort.

Furthermore, I am not sure that all the repositories under rust-lang are not using external actions. [This](https://github.com/rust-lang/rust-forge/blob/master/.github/workflows/deploy.yml) is just an example.

**Comment by simonsan** [comment]
I think it's actually important to not close this. Also in the PR I did I showed how to do it without external actions. We **don't** need to reimplement actions ourselves. A bit of copy & pasting from the PR and a bit of own work and this should be fine.

**Comment by MarcoIeni** [comment]
Yeah, the problem with copy pasting is that you have to maintain it.
From a security point of view, what's the difference with fixing the version of the github action with the release version or the commit hash for example?

**Comment by simonsan** [comment]
> From a security point of view, what's the difference with fixing the version of the github action with the release version or the commit hash for example?

I don't understand that question, could you rephrase it please? I mean a Github action has access to environment variables, if this will make it into `rust-lang` they sure want to keep track of which actions have access to these and that there are no malicious one's under it. To check/code review each of those actions is also a bit much, no?

**Comment by MarcoIeni** [comment]
> I don't understand that question, could you rephrase it please? 

Isn't copy pasting source code the same as [using a specific commit](https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-a-commits-sha-for-release-management)?
